File: backend/src/main/java/de/bucheeinfach/backend/services/abstracts/ProgramService.java
package de.bucheeinfach.backend.services.abstracts;

import de.bucheeinfach.backend.services.dtos.requests.ProgramRequest;
import de.bucheeinfach.backend.services.dtos.responses.ProgramCreatedResponse;
import de.bucheeinfach.backend.services.dtos.responses.ProgramGetAllResponse;

import java.util.List;

public interface ProgramService {
    List<ProgramGetAllResponse> getAllPrograms();
    ProgramCreatedResponse getProgramById(String id);
    ProgramCreatedResponse addProgram(ProgramRequest programRequest);
    ProgramCreatedResponse updateProgram(String id, ProgramRequest programRequest);
    ProgramCreatedResponse changeProgramStatus(String id, boolean status);
}

File: frontend/src/services/ProgramService.ts
import axios from 'axios';
import {Program} from "../types/Program.ts";

export default class ProgramService {
    getActiveProgramsSortedByNumberOfCourses() {
        return axios.get('/api/programs/active')
    }

    getAllPrograms() {
        return axios.get('/api/programs')
    }

    getProgramById(id: string) {
        return axios.get(`/api/programs/${id}`)
    }

    addProgram(program: Program) {
        return axios.post(`/api/programs`,program)
    }

    updateProgram(id: string, program: Program) {
        return axios.put(`/api/programs/${id}`, program)
    }

    changeProgramStatus(id:string, status: boolean) {
        return axios.put(`/api/programs/status/${id}?status=${status}`)
    }
}